src/main/java/graphql/execution/ResponseMapFactory.java

```diff
@@ -0,0 +1,22 @@
+package graphql.execution;
+
+import graphql.PublicSpi;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Allows to customize the concrete class {@link Map} implementation. For example, it could be possible to use
+ * memory-efficient implementations, like eclipse-collections.
+ */
+@PublicSpi
+public interface ResponseMapFactory {
+
+    /**
+     * The default implementation uses JDK's {@link java.util.LinkedHashMap}.
+     */
+    ResponseMapFactory DEFAULT = new DefaultResponseMapFactory();
+
+    Map<String, Object> create(List<String> fieldNames, List<Object> results);
```

Comment:
I feel like this should be generic

```
    <K,V> Map<K, V> create(List<K> fieldNames, List<V> results);
```

Also naming matters

The map we give back is insertion ordered.  This is important at times

So I think the `create` method should be `createInsertionOrdered(...)`

Reply:
```
List<String> fieldNames, List<Object> results
```

the names here should be generic

```
List<String> keys, List<Object> values
```

Reply:
at some time in the future we might be using this to make more "maps" in different places say

Reply:
thanks for the feedback! I covered all the points... I left out the generics because it is in contrast with a comment from @andimarek. 

I do see 2 alternatives:
- use `MapFactory` and have the generics
- use `ResponseMapFactory` and drop the generics (as they will always K=String and V=Object)

it is acceptance to keep `ResponseMapFactory` without generics by now?